src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -152,40 +154,52 @@ public function handle(AddProductToOrderCommand $command)
                 $combination
             );
 
-            $oldProducts = $cart->getProducts(true);
+            $cartComparator = new CartProductsComparator($cart);
             $this->addProductToCart($cart, $product, $combination, $command->getProductQuantity());
-            $newProducts = $cart->getProducts(true);
+            $updatedCartProducts = $cart->getProducts(true);
+
+            // We get all the modification (creation AND updates) and then filter them
```

Comment:
@jolelievre hmm, you mean:

we get all the modifications (newly added AND updated products from cart)?

Reply:
Yes, and then loop through them because even a modification of the product added by the command must be considered as a creation (to add a new OrderDetail)
But the `CartProductsComparator` can't know that, and I didn't find an elegant way to make it manage this, so I prefer to handle this specific case in this specific handler